operators/vision/draw_bounding_box.cc

```diff
@@ -0,0 +1,248 @@
+
+  // line  (3) __
+  DrawLineInHorizon(image, x_end - thickness, y_start, y_end - y_start, color, thickness);
+
+  // line  (4) --|
+  DrawLineInVertical(image, x_start, y_end - thickness, x_end - x_start, color, thickness);
+}
+
+void DrawBoxesForNumClasses(ImageView& image, const BoxArray& boxes, int64_t thickness) {
+  std::unordered_map<float, size_t> color_used;
+
+  for (size_t i = 0; i < boxes.NumBoxes(); ++i) {
+    const auto box = boxes.GetBox(i);
+    if (color_used.find(box[5]) == color_used.end()) {
+      if (color_used.size() >= KBGRColorMap.size()) {
+        // "The number of classes is larger than the number of colors in the color map.";
+        continue;
+      }
+      color_used.emplace(box[5], color_used.size());
+    }
+    const auto color = KBGRColorMap[(static_cast<int64_t>(color_used[box[5]]))];
+    DrawBox(image, box, boxes.BBoxMode(), color, thickness);
+  }
+}
+
+void DrawBoxesByScore(ImageView& image, const BoxArray& boxes, int64_t thickness) {
+  for (size_t i = 0; i < std::min<size_t>(KBGRColorMap.size(), boxes.NumBoxes()); ++i) {
+    const auto color = KBGRColorMap[(static_cast<int64_t>(i))];
+    DrawBox(image, boxes.GetBox(i), boxes.BBoxMode(), color, thickness);
+  }
+}
+
+}  // namespace
+
+void DrawBoundingBoxes::Compute(OrtKernelContext* context) {
+  // Setup inputs
+  const OrtValue* input_bgr = ort_.KernelContext_GetInput(context, 0ULL);
+  const OrtTensorDimensions dimensions_bgr(ort_, input_bgr);
+
+  if (dimensions_bgr.size() != 3 || dimensions_bgr[2] != 3) {
+    // expect {H, W, C} as that's the inverse of what decode_image produces.
+    // we have no way to check if it's BGR or RGB though
+    ORTX_CXX_API_THROW("[DrawBoundingBoxes] requires rank 3 BGR input in channels last format.", ORT_INVALID_ARGUMENT);
+  }
+
+  const OrtValue* input_box = ort_.KernelContext_GetInput(context, 1ULL);
+  const OrtTensorDimensions dimensions_box(ort_, input_box);
+  // x,y, x/w y/h, score, class
+  if (dimensions_box.size() != 2 || dimensions_box[1] != 6) {
+    ORTX_CXX_API_THROW("[DrawBoundingBoxes] requires rank 2 input and the last dim should be 6.", ORT_INVALID_ARGUMENT);
+  }
+  BoxArray boxes(dimensions_box, ort_.GetTensorData<float>(input_box), bbox_mode_);
+  int64_t image_size = dimensions_bgr[0] * dimensions_bgr[1] * dimensions_bgr[2];
+
+  // Setup output & copy to destination
+  // can we reuse the input buffer?
+  const std::vector<int64_t> output_dims(dimensions_bgr);
+  OrtValue* output_value = ort_.KernelContext_GetOutput(context, 0,
+                                                        output_dims.data(),
+                                                        output_dims.size());
+
+  auto* output_data = ort_.GetTensorMutableData<uint8_t>(output_value);
+  const auto* input_data = ort_.GetTensorData<uint8_t>(input_bgr);
+  // TODO: support batch in python side.
+  // So we are hard-coding batch size to 1 for now.
```

Comment:
What's the use case where we'd be doing batched inferences?

Reply:
Actually, I am confused why all pre-post steps do only support one batch.  It limits the ability to run multi-batches in mobile.

Does it have any motivations behind?

Reply:
In mobile scenarios when would you have a batch of inferences to perform at once?

Video maybe, but most things (took a photo, selected some text, recorded some audio) I would expect involved a batch size of 1 given the user cannot create input fast enough. 

And if you want to use NNAPI or CoreML you need to fix the batch size anyway. 

I think that might leave a scenario with a model with a fixed batch size that's processing video, and if we need to support that we could always use a Scan op around the preprocessing to handle the batch.

Reply:
> And if you want to use NNAPI or CoreML you need to fix the batch size anyway.

I see.
CPU EP can't provide any gains for multi-batches.

Reply:
CPU EP can support batches and there would most like be a performance gain. 

But need a concrete use case before we start considering whether it's worth implementing batch support in the pre/post processing, and how we want to do it. If the requirements of the theoretical use case can be met by using a Scan or Loop and the individual pre/post processing steps do not need to be batch aware, that would be significantly less work to develop/test/maintain.